### PR TITLE
Add configuration for markdown processing

### PR DESCRIPTION
**Pull request by xmorave2**

### CODE REVIEW

File: module/VuFind/src/VuFind/Service/MarkdownFactory.php [hunk 1]
```diff
@@ -63,10 +63,27 @@ class MarkdownFactory implements FactoryInterface
...
-            'html_input' => 'strip',
-            'allow_unsafe_links' => false,
+                'html_input' => $config->html_input ?? 'strip',
+                'allow_unsafe_links' => $config->allow_unsafe_links ?? false,
+                'enable_em' => $config->enable_em ?? true,
+                'enable_strong' => $config->enable_strong ?? true,
+                'use_asterisk' => $config->use_asterisk ?? true,
+                'use_underscore' => $config->use_underscore ?? true,
+                'unordered_list_markers' => isset($config->unordered_list_markers)
+                    && is_array($config->unordered_list_markers)
```
Comment (demiankatz) [inline module/VuFind/src/VuFind/Service/MarkdownFactory.php:77]: Did you test that this setting works as expected? I would expect `$config->unordered_list_markers` to be a `Zend\Config\Config` object in this context, whether or not it holds multiple values, so I'm concerned this check may never evaluate correctly. I haven't tried it, though, so perhaps the `Config` object is smarter than I think!

**Comment by demiankatz** [comment]
This makes sense to me -- just see my one question above. Thanks for continuing to develop this useful functionality!

File: module/VuFind/src/VuFind/Service/MarkdownFactory.php (hunk 1 above)
Comment (xmorave2) [reply module/VuFind/src/VuFind/Service/MarkdownFactory.php]: Thanks for catching this. Config object can't be as smart due to PHP limitations (is_array always return false on ArrayAccess). I fixed that in folowup.

Comment (demiankatz) [reply module/VuFind/src/VuFind/Service/MarkdownFactory.php]: Glad to help!

**Pull request merged**